pop-api/examples/nonfungibles/lib.rs

```diff
@@ -0,0 +1,272 @@
+	/// through minting, burning, and transferring.
+	#[ink(storage)]
+	pub struct NonFungibles {
+		/// Collection ID.
+		id: CollectionId,
+		/// Owner of the contract and collection. Set to the contract's instantiator.
+		owner: AccountId,
+	}
+
+	impl NonFungibles {
+		/// Creates a new NFT collection.
+		///
+		/// On success a [`Created`] event is emitted.
+		///
+		/// # Arguments
+		/// - `max_supply`: Maximum number of items in the collection.
+		///
+		/// # Notes
+		/// - Creating a NFT collection requires a deposit, which is taken from the contract's
+		///   balance on instantiation. The value provided at instantiation (via `payable`) must
+		///   therefore be sufficient to cover the collection deposit.
+		/// - Storage deposits are considered a best practice to mitigate against denial-of-service
+		///   attacks and state bloat.
+		/// - Destroying the collection later would refund the deposit back to the depositor.
+		#[ink(constructor, payable)]
+		pub fn new(max_supply: Option<u32>) -> Result<Self> {
+			// Get the next available collection ID.
+			let id = api::next_collection_id().map_err(Psp34Error::from)?;
+
+			let instance = Self { id, owner: Self::env().caller() };
+			let contract_id = instance.env().account_id();
+
+			// Set mint settings.
+			let mint_settings = MintSettings {
+				start_block: None,
+				end_block: None,
+				// Only the collection admin can mint.
+				mint_type: MintType::Issuer,
+				// Mint price is not set because only the contract can mint.
+				price: None,
+				default_item_settings: ItemSettings::all_enabled(),
+			};
+
+			// Create the collection.
+			let config = CollectionConfig {
+				settings: CollectionSettings::all_enabled(),
+				max_supply,
+				mint_settings,
+			};
+			// Contract is the admin of the collection.
+			api::create(contract_id, config).map_err(Psp34Error::from)?;
+			instance.env().emit_event(Created { id, admin: contract_id, max_supply });
+			Ok(instance)
+		}
+
+		/// Returns the ID of the collection.
+		#[ink(message)]
+		pub fn collection_id(&self) -> CollectionId {
+			self.id
+		}
+
+		/// Returns the number of items owned by an account.
+		#[ink(message)]
+		pub fn balance_of(&self, owner: AccountId) -> Result<u32> {
+			api::balance_of(self.id, owner).map_err(Psp34Error::from)
+		}
+
+		/// Returns the owner of an item, if any.
+		#[ink(message)]
+		pub fn owner_of(&self, item: ItemId) -> Result<Option<AccountId>> {
+			api::owner_of(self.id, item).map_err(Psp34Error::from)
+		}
+
+		/// Returns the total supply of the collection.
+		#[ink(message)]
+		pub fn total_supply(&self) -> Result<u128> {
+			api::total_supply(self.id).map_err(Psp34Error::from)
+		}
+
+		/// Mint an item.
+		///
+		/// On success a [`Transfer`] event is emitted.
+		///
+		/// # Arguments
+		/// - `to`: Account into which the item will be minted. This could be the contract's own
+		///   account identifier, for example.
```

Comment:
Creates more confusion than clarification tmo:
 ```suggestion
		/// - `to`: Account into which the item will be minted.
```